xrdp/xrdp_encoder.c

```diff
@@ -52,40 +52,46 @@ struct xrdp_encoder *APP_CC
 xrdp_encoder_create(struct xrdp_mm *mm)
 {
     struct xrdp_encoder *self;
+    struct xrdp_client_info *client_info;
     char buf[1024];
     int pid;
 
-    if (mm->wm->client_info->mcs_connection_type != 6) /* LAN */
+    client_info = mm->wm->client_info;
+
+    if (client_info->client_os_major != OSMAJORTYPE_UNIX)
     {
         return 0;
     }
-
-    if (mm->wm->client_info->bpp < 24)
+    if (client_info->mcs_connection_type != 6) /* LAN */
```

Comment:
Let's make this constant. It's not urgent but sometime in near future.
`CONNECTION_TYPE_MODEM`, `CONNECTION_TYPE_BROADBAND_LOW` ....

https://msdn.microsoft.com/en-us/library/cc240510.aspx